docs/template_plugin/tests/functional/op_reference/grid_sample.cpp

```diff
@@ -0,0 +1,475 @@
+    }
+
+    static std::string getTestCaseName(const testing::TestParamInfo<GridSampleParams>& obj) {
+        return obj.param.test_case_name;
+    }
+
+private:
+    static std::shared_ptr<Model> CreateFunction(const reference_tests::Tensor& data,
+                                                 const reference_tests::Tensor& grid,
+                                                 const op::v9::GridSample::Attributes& attributes) {
+        const auto in1 = std::make_shared<op::v0::Parameter>(data.type, data.shape);
+        const auto in2 = std::make_shared<op::v0::Parameter>(grid.type, grid.shape);
+        const auto grid_sample = std::make_shared<op::v9::GridSample>(in1, in2, attributes);
+        return std::make_shared<Model>(NodeVector{grid_sample}, ParameterVector{in1, in2});
+    }
+};
+
+constexpr auto GS_BICUBIC{op::v9::GridSample::InterpolationMode::BICUBIC};
+constexpr auto GS_BILINEAR{op::v9::GridSample::InterpolationMode::BILINEAR};
+constexpr auto GS_NEAREST{op::v9::GridSample::InterpolationMode::NEAREST};
+
+constexpr auto GS_BORDER{op::v9::GridSample::PaddingMode::BORDER};
+constexpr auto GS_REFLECTION{op::v9::GridSample::PaddingMode::REFLECTION};
+constexpr auto GS_ZEROS{op::v9::GridSample::PaddingMode::ZEROS};
+
+constexpr std::array<op::v9::GridSample::PaddingMode, 3> padding_modes{GS_ZEROS, GS_BORDER, GS_REFLECTION};
+constexpr std::array<bool, 2> align_modes{false, true};
+
+std::vector<GridSampleParams> generateNearestParamsOddDimensionsInnerGrids() {
+    std::vector<GridSampleParams> params;
+
+    reference_tests::Tensor data_odd_dims{{1, 1, 3, 5},
+                                          element::f32,
+                                          std::vector<float>{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15}};
+    reference_tests::Tensor grid_inner{
+        {1, 3, 4, 2},
+        element::f32,
+        std::vector<float>{-0.1, -0.1, -0.1, 0.1, 0.1, -0.1, 0.1, 0.1, -0.5, -0.5, -0.5, 0.5,
+                           0.5,  -0.5, 0.5,  0.5, -1., -1.,  -1., 1.,  1.,   -1.,  1.,   1.}};
+    reference_tests::Tensor output{{1, 1, 3, 4},
+                                   element::f32,
+                                   std::vector<float>{8, 8, 8, 8, 2, 12, 4, 14, 1, 11, 5, 15}};
+    for (const auto& padding : padding_modes) {
+        for (const auto align : align_modes) {
+            std::stringstream name;
+            name << "nearest_" << padding << (align ? "_align" : "_noalign") << "_odd_dims_inner";
+            params.emplace_back(data_odd_dims,
+                                grid_inner,
+                                op::v9::GridSample::Attributes{align, GS_NEAREST, padding},
+                                output,
+                                name.str());
+        }
+    }
+    return params;
+}
+
+std::vector<GridSampleParams> generateNearestParamsOddDimensionsOuterGrids() {
+    std::vector<GridSampleParams> params;
+    reference_tests::Tensor data_odd_dims{{1, 1, 3, 5},
```

Comment:
Could you add a test with batch size (N) dim != 1?

Reply:
done